lib/builtin.pm

```diff
@@ -106,6 +106,14 @@ The following bundles currently exist:
     :5.40      true false weaken unweaken is_weak blessed refaddr reftype
                ceil floor is_tainted trim indexed
 
+=head2 Function Optimisations
+
+Various optimisations that apply to many functions in the L<builtin> package
+would be no longer be performed if the functions are ever replaced or changed,
+such as by assignment into glob references.  It is suggested not to do this,
+so that the optimiser can correctly work out that intention of code which
+calls these functions.
+
```

Comment:
"would be no longer be" -> "would no longer be".

"work out that intention" -> "work out the intention" (maybe?)

The whole thing feels very passive voice.

Maybe:

```
There are a number of optimisations that apply to functions in the L<builtin> package. If you replace or override these functions, the optimisations may not take effect. Do so with caution.
```

Reply:
Ahyes, some better words.